frontend/javascripts/libs/datasource.types.js

```diff
@@ -4,8 +4,9 @@
 // The `flow2schema` node module has been used for conversion.
 // Please note that some manual changes to the schema are required.
 
-// Add minItems=3 and maxItems=3
-type Vector3 = Array<number>;
+type Vector3 = [number, number, number];
+
+type Vector2 = [number, number];
```

Comment:
I think defining the Vectors as tuples is more accurate as it defines the length of the array